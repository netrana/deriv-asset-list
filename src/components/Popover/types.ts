import { TooltipPlacement } from "antd/lib/tooltip";

export interface Props{
    content: JSX.Element;
    placement?: TooltipPlacement;
    trigger?: string; 
}