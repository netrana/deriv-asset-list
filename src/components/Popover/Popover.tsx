import { FC } from "react";
import { Popover as AntdPopover } from "antd";

import { Props } from './types';

export const Popover: FC<Props> = (props) => {
    const { content, children, placement = 'bottomLeft', trigger='click' } = props

  return (
    <AntdPopover placement={placement} content={content} trigger={trigger}>
      {children}
    </AntdPopover>
  );
};

export default Popover;
