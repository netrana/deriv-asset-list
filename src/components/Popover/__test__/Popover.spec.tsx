import render from "react-test-renderer";
import Popover from "../Popover";

describe("Popover component test", () => {
  describe("snapshot", () => {
    it("renders correctly", () => {
      const tree = render
        .create(<Popover content={<div>Test</div>} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
