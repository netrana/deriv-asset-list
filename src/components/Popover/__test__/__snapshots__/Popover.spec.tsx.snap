// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component test snapshot renders correctly 1`] = `
<span
  onClick={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
/>
`;
